subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/gradle/ConsumerProvidedTask.java

```diff
@@ -19,12 +19,13 @@
 import com.google.common.collect.Sets;
 import org.gradle.tooling.internal.consumer.converters.TaskNameComparator;
 
+import java.io.Serializable;
 import java.util.SortedSet;
 
 /**
  * A consumer-side implementation of {@link org.gradle.tooling.model.Task}.
  */
-public class ConsumerProvidedTask implements TaskListingLaunchable {
+public class ConsumerProvidedTask implements TaskListingLaunchable, Serializable {
```

Comment:
It is used in compatibility mappings. These are now also run inside build actions. The result of a build action needs to be serialized back to the client.